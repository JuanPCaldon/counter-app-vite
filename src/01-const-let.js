

// variables 

const nombre = ' juan pablo' ;

let apellidos = 'pinilla caldon';


console.log(nombre , apellidos);




