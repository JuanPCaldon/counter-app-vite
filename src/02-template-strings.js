

//  templetate strings


//const nombre = 'juan';
//const apellido= 'pinilla ';

//const nombre_completo = nombre + ' ' + apellido;

//const nombre_completo = ` ${nombre}  ${apellido}  `;
//console.log(nombre_completo);




export function getsaludo(nombre){

  return 'hola'+' ' + nombre;

}

