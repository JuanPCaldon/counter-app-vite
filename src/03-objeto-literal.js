
//


const persona = {
    nombre: 'tony',
    apellido : 'stark',
    edad: 45, 
    direccion : {
       ciudad : ' new york',
       zip: '555563532' ,
       lat: '4654622.45' ,
       lng: '45452',


    }

   
};


console.log( persona );


const persona2 = {...persona };


 persona2.nombre = 'peter';