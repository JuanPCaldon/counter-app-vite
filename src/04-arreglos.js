

const arreglo = [1,2,3,4 ];


let arreglo2 = [...arreglo , 5];


const arreglo3 = arreglo2.map( function(numero){
   
    return numero * 2; 


}); 

console.log(arreglo3);