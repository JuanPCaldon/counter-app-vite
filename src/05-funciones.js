

//funciones en js

const saludar= function(nombre){
  
    
    return `hola ${ nombre}`;


};

//const saludar2 = (nombre)=>{

  //  return `hola ${nombre}`;

//};


//const saludar3 = (nombre) => `hola, ${nombre}`



//console.log(saludar('goku'));
//console.log(saludar2('vegeta'));
//console.log(saludar3('goku'));



export const getuser = () =>{
return{
 Uid:'ABC13274',
 username: 'el_papi41564'

}

};
//console.log( getuser());



//tarea 

//2. retornar  un objeto implicito
//3. pruebas
export const  getusuarioactivo = (nombre)=> ({
    
     uid: 'ABC125',
     username: nombre,

});

//const usuarioactivo = getusuarioactivo('fernando');
//console.log( usuarioactivo);

