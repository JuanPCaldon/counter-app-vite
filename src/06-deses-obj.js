
//desestructuracion
//asignacion desestructurante

const persona = {

    nombre: 'tony',
    edad: 45 ,
    clave: 'ironman',
};


//const {nombre, edad , clave } = persona ;

//console.log(nombre);
//console.log(edad);
//console.log(clave);

const extraerpersona = ({clave ,nombre,edad, rango = 'capitan'}) => {

  //  console.log(nombre,edad,rango);

  return {

    nombreClave : clave,
    anios: edad,

    latlng:{
        lat:54.285,
        lng:-12.57
 
    }
  }
  

}

const {nombreClave , anios, latlng:{lat,lng}} = extraerpersona(persona);



console.log(nombreClave , anios);
console.log(lat,lng);

