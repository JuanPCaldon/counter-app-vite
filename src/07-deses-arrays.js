
//desestructuracion de arreglos
//
//const personajes =[ 'goku','vegeta', 'trunks'];


//const [,p1] =personajes;

//console.log(p1);


export const retornaArreglo = ()=>{

    return ['ABC',123];

};



//const [letras , numeros ] =retornaArreglo();

//console.log( letras , numeros);


//Tarea
// 1. el primer valor del arr se llamara nombre
// 2. se llamara setnombre

//const tarea1 = (nombre ) =>{

//    return [ nombre , ()=> console.log('hola mundo') ]

//} 




//const [nombre , setnombre ] = tarea1('goku');

//console.log(nombre);
//setnombre();

