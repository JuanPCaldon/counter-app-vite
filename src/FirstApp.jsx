import  PropTypes  from 'prop-types';

//fragmento <> </>

const NewMesage={
 
    message : 'hola mundo',
    tittle :  'juan pablo',


};   

//const imprimir= function prueba(){

//    return (`hola soy juan ${ 2 + 2}`);



//};

export  function FirstApp ( {tittle, subtittle})  {
  
   
    
   
     
     return ( 
        <>

        <h1 data-testid="test-tittle"> {tittle} </h1>

        {/* <h1>{ JSON.stringify( NewMesage) }</h1> */}
        <p> {subtittle } </p>
        
        </>
        
       
     );


}

FirstApp.propTypes = {

    tittle: PropTypes.string.isRequired,
    //subtittle:PropTypes.number.isRequired,


}

FirstApp.defaultProps = {
 tittle:'no hay titulo',
 //subtittle:'no hay subtitulo',
 name: 'juan pinilla',
  


}