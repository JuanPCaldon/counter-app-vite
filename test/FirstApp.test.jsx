import { render } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';


 describe('pruebas en <FirstApp/>',()=>{

  /*   test('debe de hacer match con el snapshot',()=>{

     
       const tittle = 'hola soy goku';
       const{container}= render( <FirstApp tittle={ tittle}  /> )
        

       expect(container).toMatchSnapshot();

    });
      */

    test('debe mostrar el titulo en un h1',()=>{


        const tittle = 'hola soy goku';
        const{container,getByText,getByTestId}= render( <FirstApp tittle={ tittle}  /> )
        //expect(getByText(tittle)).toBeTruthy();
        expect(getByTestId('test-tittle').innerHTML).toContain(tittle);
        /* const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(tittle);
 */






    });


  

      



} );