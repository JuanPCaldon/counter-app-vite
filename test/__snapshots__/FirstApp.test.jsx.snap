// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
     
    hola soy goku
     
  </h1>
  <p>
     
     
  </p>
</div>
`;
